server/src/main/java/com/freded/file/server/controller/TaskFileUploadService.java

```
@@ -0,0 +1,108 @@
+      throw new CustomWebApplicationException("Task not found with ID", 400);
+    }
+
+    if (taskFileUploadDTO.getFileUpload() == null) {
+      throw new CustomWebApplicationException("No file uploaded", 400);
+    }
+  }
+
+  private String extractAndSanitizeFileName(FileUpload fileUpload, TaskFileUploadDTO taskFileUploadDTO) {
+    String originalFileName = fileUpload.fileName();
+    if (originalFileName == null || originalFileName.isEmpty()) {
+      originalFileName = taskFileUploadDTO.getFileName();
+    }
+    return sanitizeFileName(originalFileName);
+  }
+
+  private String buildObjectName(String username, String taskId, String fileName) {
+    return String.format("user/%s/%s/%s", username, taskId, fileName);
+  }
+
+  private void uploadToMinio(FileUpload fileUpload, String objectName, String fileType) throws Exception {
+    Path filePath = fileUpload.uploadedFile();
+    try (InputStream inputStream = Files.newInputStream(filePath)) {
+      minioUploadService.uploadFile(objectName, inputStream, fileUpload.size(), fileType);
+    }
+  }
+
+  private TaskFileEntity createFileEntity(
+      String fileName, String fileType, String objectName, String uploadedBy, String taskId) {
+    TaskFileEntity fileEntity = new TaskFileEntity();
+    fileEntity.setFileName(fileName);
+    fileEntity.setFileType(fileType);
+    fileEntity.setObjectName(objectName);
+    fileEntity.setUploadedBy(uploadedBy);
+    fileEntity.setTaskId(taskId);
+    return fileEntity;
+  }
+
+  private TaskFileDTO saveAndReturnDTO(TaskFileEntity fileEntity) {
+    entityManager.persist(fileEntity);
+    LOG.info("Created file dto with ID: " + fileEntity.getId());
+    return taskFileMapper.toDTO(fileEntity);
+  }
+
+  private void handleError(String fileName, Exception ex) {
+    LOG.error("Failed to save file: " + fileName, ex);
+  }
+
+  private String sanitizeFileName(String fileName) {
+    if (fileName == null || fileName.isEmpty()) {
+      return "unknown_file";
```

Why should you write unknown file here? Then, a file called unknown_file could be read or written (or?). The name should never be null or? You could throw an exception here or even return null and throw an exception later.